### PR TITLE
Improve sig handler

### LINKED ISSUE #8533: [Feature] Print line number during core could help find the root cause
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Currently doris will print core stack at log/be.out like this, but the line number is missed.
105 Minidump created at: /data/ygl/code/github/incubator-doris/test_output/be2/minidump/58c19530-492b    -40a5-2610f9a7-649c2eeb.dmp
106 *** Aborted at 1647611812 (unix time) try "date -d @1647611812" if you are using GNU date ***
107 PC: @     0x7f3988ffa561 doris::OlapScanner::prepare()
108 *** SIGSEGV (@0x8) received by PID 28783 (TID 0x7f3951798700) from PID 8; stack trace: ***
109     @     0x7f398a558672 google::(anonymous namespace)::FailureSignalHandler()
110     @     0x7f3986224270 (unknown)
111     @     0x7f3988ffa561 doris::OlapScanner::prepare()
112     @     0x7f3988fb57c4 doris::OlapScanNode::start_scan_thread()
113     @     0x7f3988fb8cca doris::OlapScanNode::start_scan()
114     @     0x7f3988fb923e doris::OlapScanNode::get_next()
115     @     0x7f398907c711 doris::PartitionedAggregationNode::open()
116     @     0x7f398897f4fd doris::PlanFragmentExecutor::open_internal()
117     @     0x7f3988980a9d doris::PlanFragmentExecutor::open()
118     @     0x7f39888f1635 doris::FragmentExecState::execute()
119     @     0x7f39888f5efa doris::FragmentMgr::_exec_actual()
120     @     0x7f39888ff40c std::_Function_handler<>::_M_invoke()
121     @     0x7f3988aa7459 doris::ThreadPool::dispatch_thread()
122     @     0x7f3988aa0a4f doris::Thread::supervise_thread()
123     @     0x7f3985fdae25 start_thread
124     @     0x7f39862e735d __clone
125     @                0x0 (unknown)

If line number is added to the stack, it will help debug.

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yiguolei**
# Proposed changes

Issue Number: close #8533 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by yiguolei** [comment]
There will be more detail core stack info: 

- Detail signal explanation 
- Add line number to core stack

`*** Aborted at 1647785336 (unix time) try "date -d @1647785336" if you are using GNU date *** *** SIGSEGV address not mapped to object (@0x8) received by PID 8750 (TID 0x7fe2423da700) from PID 8; stack trace: *** 0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /data/ygl/code/github/incubator-doris/be/src/common/signal_handler.h:423 1# 0x00007FE255D23270 in /lib64/libc.so.6 2# doris::Tablet::Tablet(std::shared_ptr<doris::TabletMeta>, doris::DataDir*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at /data/ygl/code/github/incubato r-doris/be/src/olap/tablet.cpp:86 3# doris::Tablet::create_tablet_from_meta(std::shared_ptr<doris::TabletMeta>, doris::DataDir*) at /data/ygl/code/github/incubator-doris/be/src/olap/tablet.cpp:61 4# doris::TabletManager::load_tablet_from_meta(doris::DataDir*, long, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, bool, bool, bool) at /data/ygl /code/github/incubator-doris/toolchain/include/c++/11/bits/shared_ptr.h:122 5# std::_Function_handler<bool (long, long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&), doris::DataDir::load()::{lambda(long, int, std::__cxx11::basic_str ing<char, std::char_traits<char>, std::allocator<char> > const&)#2}>::_M_invoke(std::_Any_data const&, long&&, long&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > c onst&) at /data/ygl/code/github/incubator-doris/toolchain/include/c++/11/bits/std_function.h:291 6# std::_Function_handler<bool (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&), doris::TabletMetaManager::traverse_headers(doris::OlapMeta*, std::function<bool (long, long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)> const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)::{lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::b asic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}>::_M_invoke(std::_Any_data const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, s td::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at /data/ygl/code/github/incubator-doris/toolchain/include/c++/11/bits/std_function.h:291 7# doris::OlapMeta::iterate(int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::function<bool (std::__cxx11::basic_string<char, std::char_traits<char>, s td::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)> const&) at /data/ygl/code/github/incubator-doris/be/src/olap/olap_meta.cpp:183 8# doris::TabletMetaManager::traverse_headers(doris::OlapMeta*, std::function<bool (long, long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)> const&, std::_ _cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at /data/ygl/code/github/incubator-doris/be/src/olap/tablet_meta_manager.cpp:134 9# doris::DataDir::load() at /data/ygl/code/github/incubator-doris/be/src/olap/data_dir.cpp:453 10# std::thread::_State_impl<std::thread::_Invoker<std::tuple<doris::StorageEngine::load_data_dirs(std::vector<doris::DataDir*, std::allocator<doris::DataDir*> > const&)::{lambda()#1}> > >::_M_run() a t /data/ygl/code/github/incubator-doris/toolchain/include/c++/11/bits/std_thread.h:211 11# execute_native_thread_routine in /data/ygl/code/github/incubator-doris/test_output/be/lib/palo_be 12# start_thread in /lib64/libpthread.so.0 13# clone in /lib64/libc.so.6`

**Comment by yiguolei** [comment]
Build will failed, because the default build process will try to download libbacktrace from baidu bos, while it is on github. Somebody in baidu could help me upload it to bce bos.

File: thirdparty/vars.sh [hunk 1]
```diff
@@ -52,6 +52,12 @@ export REPOSITORY_URL=https://doris-thirdparty-repo.bj.bcebos.com/thirdparty
 # of all thirdparties
 #####################################################
 
+# libbacktrace
+LIBBACKTRACE_DOWNLOAD="https://github.com/Doris-Extras/libbacktrace/archive/refs/tags/libbacktrace-1.0.zip"
```
Comment (yangzhg) [inline thirdparty/vars.sh:56]: why not use the origin repo ?

File: be/src/common/signal_handler.h [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+      : buffer_(buffer),
+        cursor_(buffer),
+        end_(buffer + size) {
+  }
+
+  // Returns the number of bytes written in the buffer.
+  std::size_t num_bytes_written() const { return static_cast<std::size_t>(cursor_ - buffer_); }
+
+  // Appends string from "str" and updates the internal cursor.
+  void AppendString(const char* str) {
```
Comment (yangzhg) [inline be/src/common/signal_handler.h:196]: Why  use Camel Case here?

File: thirdparty/vars.sh [hunk 3]
```diff
@@ -372,7 +378,8 @@ SIMDJSON_SOURCE=simdjson-1.0.2
 SIMDJSON_MD5SUM="5bb34cca7087a99c450dbdfe406bdc7d"
 
 # all thirdparties which need to be downloaded is set in array TP_ARCHIVES
-export TP_ARCHIVES="LIBEVENT
+export TP_ARCHIVES="LIBBACKTRACE
```
Comment (yangzhg) [inline thirdparty/vars.sh:381]: if you add a lib, and use it in CMake by default , you should check it at build.sh

**Comment by yangzhg** [comment]
> .

File: be/src/common/signal_handler.h (hunk 2 above)
Comment (yiguolei) [reply be/src/common/signal_handler.h:196]: I copied glog's signal handler and modify it. I am not sure whether should I modify the code format... If it is required, I could modify it.

File: thirdparty/vars.sh [hunk 4]
```diff
@@ -371,6 +371,12 @@ SIMDJSON_NAME=simdjson-1.0.2.tar.gz
 SIMDJSON_SOURCE=simdjson-1.0.2
 SIMDJSON_MD5SUM="5bb34cca7087a99c450dbdfe406bdc7d"
 
+# libbacktrace
+LIBBACKTRACE_DOWNLOAD="https://codeload.github.com/ianlancetaylor/libbacktrace/zip/2446c66076480ce07a6bd868badcbceb3eeecc2e"
+LIBBACKTRACE_NAME=2446c66076480ce07a6bd868badcbceb3eeecc2e.zip
```
Comment (yangzhg) [inline thirdparty/vars.sh:376]:
```suggestion
LIBBACKTRACE_NAME=libbacktrace-2446c66076480ce07a6bd868badcbceb3eeecc2e.zip
```

**Review by yangzhg — APPROVED** [review]
LGTM

**Comment by Gabriel39** [comment]
@yiguolei should we re-build docker image to add libbacktrace.a into the new image?

**Comment by yiguolei** [comment]
@Gabriel39 You are right. @yangzhg  Said he will rebuild the docker img and upload it.

**Pull request merged**